### PR TITLE
Cleanup Pylint singleton issues

### PR DESCRIPTION
**Pull request by drawlerr**
There are a number of 1-off pylint warnings that I've fixed and bundled in a single PR for simplicity.
Addressed:
 -W1202
 -W0706
 -W0702
 -W0127
 -R1723
 -R1714
 -R1706
 -R1702
 -E0911
 -E0601

### CODE REVIEW

File: esrally/actor.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def bootstrap_actor_system(try_join=False, prefer_local_only=False, local_ip=Non
                 coordinator_ip = None
                 local_ip = None
         else:
-            if system_base != "multiprocTCPBase" and system_base != "multiprocUDPBase":
+            if system_base not in ('multiprocTCPBase', 'multiprocUDPBase'):
```
Comment (danielmitterdorfer) [inline esrally/actor.py]: Can we please keep the double quotes?

File: esrally/mechanic/supplier.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def _supply_requirements(sources, distribution, build, plugins, revisions, distr
                 # * --pipeline=from-sources-skip-build --distribution-version=X.Y.Z where the plugin should not be built but ES should be
                 #   a distributed version.
                 # * --distribution-version=X.Y.Z --revision="my-plugin:abcdef" where the plugin should be built from sources.
-                plugin_needs_build = (sources and build) or distribution
+                plugin_needs_build = build if sources else distribution
```
Comment (danielmitterdorfer) [inline esrally/mechanic/supplier.py]: Personally I find the previous styler easier to read and I vote for keeping this and ignoring pylint's suggestion.

File: esrally/track/track.py [hunk 3]
```diff
@@ -450,63 +450,64 @@ def admin_op(self):
     @classmethod
     def from_hyphenated_string(cls, v):
```
Comment (danielmitterdorfer) [inline esrally/track/track.py]: I think `R0911` is unjustified for a method like this. I vote to ignore it here and keep the old style.

**Comment by danielmitterdorfer** [comment]
@drawlerr can you please:

* Reference this PR in all issues mentioned in the checklist in #838 so we know what's fixed by which PR? You should do this already now before the PR is merged.
* Add a `Relates #838` in your merge commit?

File: esrally/actor.py (hunk 1 above)
Comment (drawlerr) [reply esrally/actor.py]: Oops, I was just taking the Pylint suggestion as-is.  Changing back to double quotes.

**Pull request merged**